### ISSUE TITLE
Adding Tests + CI

### ISSUE THREAD

**Issue by bhuvy2** [illinois-cs241/broadway-grader#22]
The CI can be mostly copied over from broadway-api, but it'd be nice to have some assurance checks on each commit.

**Comment by ayushr2** [comment]
I think this issue needs to be shifted to [Chain Link](https://github.com/illinois-cs241/chainlink). Broadway grader just communicates with the API and passes on the job to chainlink to run. @nmagerko wrote some [tests for chain link](https://github.com/illinois-cs241/chainlink/tree/master/tests/integration) which tests what we want to test.

What would we want to test on the broadway-grader side of things?

**Comment by bhuvy2** [comment]
There is also a discussion of whether we want to merge chain link. Assuming we want to keep the different, we should just sanity test heartbeating and whether chainlink calls the right funcs

**Comment by ayushr2** [comment]
Yep, I think we should keep them separate. Chainlink is meant to be a public python library for running a chain of containers. This project is more Broadway specific.

**Comment by bhuvy2** [comment]
I don't want to get into the whole debate of what warrants a package
(I strongly side on [functions or few functions don't warrant packages re: the left-pad dabacle](https://www.davidhaney.io/npm-left-pad-have-we-forgotten-how-to-program/), and chain-link in spirit is a different functionality, but is not a collection of functions or otherwise.

Either way, mocking should be added.

**Comment by nmagerko** [comment]
I don't like the idea of making chainlink part of this at all. Discussing left-pad here feels like a pretty big over-exaggeration (not to mention that we're not using Node, which has an entirely different set of ideas about modules/packages).

**Comment by nmagerko** [comment]
As for testing, I think this project might need some refactoring first (this project can be tested in a way that is separate from how chainlink tests). We should at least be able to test that a job is submitted to some black box (e.g. `chainlink`) to run, and that the results we get back from the `broadway-grader` routine that does that makes sense.

**Comment by nmagerko** [comment]
The actual execution of running containers in sequence should be tested more throughly, but that is not the responsibility of this project. I'd say the needs to be a `chainlink` issue.
